client/handle_http_test.go

```diff
@@ -551,3 +552,620 @@ func TestFullUpload(t *testing.T) {
+	viper.Set("Logging.LogLocation", output.Name())
+
+	originDir, err := os.MkdirTemp("", "Origin")
+	assert.NoError(f.T, err)
+	f.OriginDir = originDir
+
+	// Change the permissions of the temporary origin directory
+	permissions = os.FileMode(0777)
+	err = os.Chmod(originDir, permissions)
+	require.NoError(f.T, err)
+
+	viper.Set("Origin.ExportVolume", originDir+":/test")
+	viper.Set("Origin.Mode", "posix")
+	viper.Set("Origin.EnableFallbackRead", true)
+	// Disable functionality we're not using (and is difficult to make work on Mac)
+	viper.Set("Origin.EnableCmsd", false)
+	viper.Set("Origin.EnableMacaroons", false)
+	viper.Set("Origin.EnableVoms", false)
+	viper.Set("Origin.EnableWrite", true)
+	viper.Set("TLSSkipVerify", true)
+	viper.Set("Server.EnableUI", false)
+	viper.Set("Registry.DbLocation", filepath.Join(f.T.TempDir(), "ns-registry.sqlite"))
+	viper.Set("Xrootd.RunLocation", tmpPath)
+
+	err = config.InitServer(ctx, modules)
+	require.NoError(f.T, err)
+
+	viper.Set("Registry.RequireOriginApproval", false)
+	viper.Set("Registry.RequireCacheApproval", false)
+
+	f.FedCancel, err = launchers.LaunchModules(ctx, modules)
+	if err != nil {
+		f.T.Fatalf("Failure in fedServeInternal: %v", err)
+	}
+
+	desiredURL := param.Server_ExternalWebUrl.GetString() + "/.well-known/openid-configuration"
+	err = server_utils.WaitUntilWorking(ctx, "GET", desiredURL, "director", 200)
+	require.NoError(f.T, err)
+
+	httpc := http.Client{
+		Transport: config.GetTransport(),
+	}
+	resp, err := httpc.Get(desiredURL)
+	require.NoError(f.T, err)
+
+	assert.Equal(f.T, resp.StatusCode, http.StatusOK)
+
+	responseBody, err := io.ReadAll(resp.Body)
+	require.NoError(f.T, err)
+	expectedResponse := struct {
+		JwksUri string `json:"jwks_uri"`
+	}{}
+	err = json.Unmarshal(responseBody, &expectedResponse)
+	require.NoError(f.T, err)
+
+	f.Cancel = cancel
+	f.ErrGroup = egrp
+}
+
+func (f *FedTest) Teardown() {
+	os.RemoveAll(f.TmpPath)
+	os.RemoveAll(f.OriginDir)
+	f.Cancel()
+	f.FedCancel()
+	assert.NoError(f.T, f.ErrGroup.Wait())
+	viper.Reset()
+}
+
+// A test that spins up a federation, and tests object get and put
+func TestGetAndPutAuth(t *testing.T) {
+	// Create instance of test federation
+	viper.Reset()
+	fed := FedTest{T: t}
+	fed.Spinup()
+	defer fed.Teardown()
+
+	// Other set-up items:
+	testFileContent := "test file content"
+	// Create the temporary file to upload
+	tempFile, err := os.CreateTemp(t.TempDir(), "test")
+	assert.NoError(t, err, "Error creating temp file")
+	defer os.Remove(tempFile.Name())
+	_, err = tempFile.WriteString(testFileContent)
+	assert.NoError(t, err, "Error writing to temp file")
+	tempFile.Close()
+
+	// Create a token file
+	tokenConfig := utils.TokenConfig{
+		TokenProfile: utils.WLCG,
+		Lifetime:     time.Minute,
+		Issuer:       param.Origin_Url.GetString(),
+		Audience:     []string{param.Origin_Url.GetString()},
+		Subject:      "origin",
+	}
+	tokenConfig.AddRawScope("storage.read:/ storage.modify:/")
```

Comment:
I think Justin's recent PR added support for WLCG token scopes as enumerated type. Could you update this? Here is a reference: https://github.com/PelicanPlatform/pelican/blob/5b921c42bdb965f6413d2f93fc874a993a611dc3/client/handle_http_test.go#L391